dev/scripts/build.ts

```diff
@@ -3,46 +3,139 @@ import { join } from "node:path"
 import { readJson, writeJson } from "../attest/src/fs.js"
 import { shell } from "../attest/src/shell.js"
 import { repoDirs } from "./common.js"
+import { rewritePaths } from "./overwrite.js"
+import type { ReplacementDictionary } from "./overwrite.js"
 
 const packageRoot = process.cwd()
 const outRoot = join(packageRoot, "dist")
 const packageJson = readJson(join(packageRoot, "package.json"))
+const tempTsConfigBaseName = "tsconfig.temp"
+const tempTsConfigPath = join(packageRoot, `${tempTsConfigBaseName}.json`)
 
-const buildFormat = (module: "commonjs" | "esnext") => {
-    const outDir = join(outRoot, module === "commonjs" ? "cjs" : "mjs")
-    writeJson(tempTsConfig, {
-        ...tsConfigData,
-        include: ["src"],
+const nuke = (target: string) =>
+    rmSync(target, { recursive: true, force: true })
+const clone = (from: string, to: string): void =>
+    cpSync(from, to, { recursive: true, force: true })
+
+const writeManifest =
+    (overrides: Record<string, unknown>) =>
+    (sourceDir: string, targetDir: string) => {
+        const manifest = readJson(join(sourceDir, "package.json"))
+        writeJson(join(targetDir, "package.json"), {
+            ...manifest,
+            ...overrides
+        })
+    }
+
+const Sources = {
```

Comment:
Can we reuse the paths from `repoDirs` here instead?

Reply:
Not here -- in this file, the root changes in several places, sometimes dynamically.